src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -167,28 +192,13 @@ VideoDecoder::BatchDecodedOutput::BatchDecodedOutput(
     const VideoStreamDecoderOptions& options,
     const StreamMetadata& metadata)
     : ptsSeconds(torch::empty({numFrames}, {torch::kFloat64})),
-      durationSeconds(torch::empty({numFrames}, {torch::kFloat64})) {
-  if (options.dimensionOrder == "NHWC") {
-    frames = torch::empty(
-        {numFrames,
-         options.height.value_or(*metadata.height),
-         options.width.value_or(*metadata.width),
-         3},
-        {torch::kUInt8});
-  } else if (options.dimensionOrder == "NCHW") {
-    frames = torch::empty(
-        {numFrames,
-         3,
-         options.height.value_or(*metadata.height),
-         options.width.value_or(*metadata.width)},
-        torch::TensorOptions()
-            .memory_format(torch::MemoryFormat::ChannelsLast)
```

Comment:
I am wondering if using this is identical to permuting a NHWC tensor at the end. I am not 100% sure. Do you?

Reply:
I meant in terms of performance, is it identical? Specifically this is a bit concerning:

![Uploading image.png…]()


We really should have a batch benchmark because the whole point of batch is to do things in a performant way for many frames.

Reply:
Sorry I don't understand what you mean. What image did you mean to link to?

Note that this PR should e strictly more efficient:

> Batched output tensors are now always created as NHWC. They are converted to NCHW in one single step, instead of converting HWC sub-tensors N times.

Reply:
Sorry the image wasn't uploaded properly:

<img width="872" alt="image" src="https://github.com/user-attachments/assets/06e1d83e-79ec-4dfb-8a60-6c6a8ed25812">

I am not sure about the performance implications of doing a permute instead of `.to(channels_last)`

Reply:
Sorry I still don't really understand where you're coming from. Can you please share a link? Is this relevant for this PR? 

Again, the change involved in this PR is:

> Batched output tensors are now always created as NHWC. They are converted to NCHW in one single step, instead of converting HWC sub-tensors N times

Reply:
Thanks for the link.

We're not concerned about memory format (contiguous vs channels-last) in this PR. This is a related but distinct concern to the dimension order.

Reply:
Link in the edited comment above. It's from the channels-last page.

I am not 100% sure if creating a NHWC and permuting it is the same as creating a NCHW with channels-last and working with that. The code that you deleted was doing the latter.

A benchmark may show a difference -- or not. Do you know?